osquery/TPipe.py

```diff
@@ -0,0 +1,225 @@
+
+        if err:
+            raise TTransportException(
+                type=TTransportException.UNKNOWN,
+                message='TPipe read failed with GLE={}'.format(err))
+        if len(buff) == 0:
+            raise TTransportException(
+                type=TTransportException.END_OF_FILE,
+                message='TPipe read 0 bytes')
+        return buff
+
+    def write(self, buff):
+        if not self.is_open():
+            raise TTransportException(
+                type=TTransportException.NOT_OPEN,
+                message='Called read on non-open pipe')
+
+        bytesWritten = None
+        try:
+            (_, bytesWritten) = win32file.WriteFile(self._handle, buff, None)
+        except Exception as e:
+            raise TTransportException(
+                type=TTransportException.UNKNOWN,
+                message='Failed to write to named pipe: ' + e.message)
+
+        if bytesWritten != len(buff):
+            raise TTransportException(
+                type=TTransportException.UNKNOWN,
+                message='Failed to write complete buffer to named pipe')
+
+    def flush(self):
+        if self.is_open():
+            win32file.FlushFileBuffers(self._handle)
+
+
+class TPipeServer(TPipeBase, TServerTransportBase):
+    """Named pipe implementation of TServerTransport"""
+
+    def __init__(self,
+                 pipe_name,
+                 buff_size=4096,
+                 max_conns=255,
+                 max_conn_attempts=5):
+        """
+        Initialize a TPipe client
+
+        @param pipe_name(string)  The path of the pipe to connect to.
+        @param buff_size(int)  The size of read/write buffers to use.
+        @param max_conns(int)  Maximum number of simultaneous connections.
+        @param max_conn_attempts(int)  Maximum number of connection attempts
+        """
+        self._pipe_name = pipe_name
+        self._buff_size = buff_size
+        self._max_conns = max_conns
+        self._max_conn_attempts = max_conn_attempts
+        self._handle = None
+
+        # Overlapped event for Windows Async IO
+        self._overlapped = pywintypes.OVERLAPPED()
+        self._overlapped.hEvent = win32event.CreateEvent(None, 0, 0, None)
+
+    # Create a new named pipe if one doesn't already exist
+    def listen(self):
+        if self._handle is None:
+            self.create_named_pipe()
+
+    def accept(self):
+        if self._handle is not None:
+            self.create_named_pipe()
+
+        self.initiate_named_connect()
+        result = TPipe(self._pipe_name)
+        result.set_handle(self._handle)
+
+        return result
+
+    def create_named_pipe(self):
+        openMode = win32pipe.PIPE_ACCESS_DUPLEX | win32file.FILE_FLAG_OVERLAPPED
+        pipeMode = win32pipe.PIPE_TYPE_BYTE | win32pipe.PIPE_READMODE_BYTE
+
+        saAttr = pywintypes.SECURITY_ATTRIBUTES()
+        saAttr.SetSecurityDescriptorDacl(1, None, 0)
+
+        self._handle = win32pipe.create_named_pipe(
```

Comment:
This causes an error since create_named_pipe doesn't exist in the version of pywin32 I have. 

Changing it to  win32pipe.CreateNamedPipe fixed it.

Reply:
Note this was using the latest pypiwin32.